zlibWrapper/zstd_zlibwrapper.c

```diff
@@ -67,29 +69,52 @@ static int g_useZSTD = ZWRAP_USE_ZSTD;   /* 0 = don't use ZSTD */
 
 void useZSTD(int turn_on) { g_useZSTD = turn_on; }
 
-int isUsingZSTD() { return g_useZSTD; }
+int isUsingZSTD(void) { return g_useZSTD; }
 
-const char * zstdVersion() { return ZSTD_VERSION_STRING; }
+const char * zstdVersion(void) { return ZSTD_VERSION_STRING; }
 
 ZEXTERN const char * ZEXPORT z_zlibVersion OF((void)) { return zlibVersion();  }
 
 
+static void* ZWRAP_allocFunction(void* opaque, size_t size)
+{
+    z_streamp strm = (z_streamp) opaque;
+    void* address = strm->zalloc(strm->opaque, 1, size);
+  /*  printf("ZWRAP alloc %p, %d \n", address, (int)size); */
+    return address;
+}
+
+static void ZWRAP_freeFunction(void* opaque, void* address)
+{
+    z_streamp strm = (z_streamp) opaque;
+    strm->zfree(strm->opaque, address);
+   /* if (address) printf("ZWRAP free %p \n", address); */
+}
+
+
 
 /* *** Compression *** */
 
 typedef struct {
     ZBUFF_CCtx* zbc;
     size_t bytesLeft;
     int compressionLevel;
+    z_stream allocFunc; /* copy of zalloc, zfree, opaque */
```

Comment:
OK, so it's a copy, not a reference to parent.

If `zalloc`, `zfree` or `opaque` are changed after `deflateInit()`, the changes will not be taken in consideration.
No allocation is needed outside of `deflateInit()` anyway,  but `zfree` will be needed at the end. Hopefully, it seems logical to not "change" `zfree` nor `opaque` between `deflateInit()` and `deflateEnd()`. Maybe a documentation warning on this fact would help.

It also means that usage of `ZWRAP_CCtx.allocFun` is fairly limited. Just once, when calling `ZWRAP_createCCtx()`.

My own choice would have been to create a structure which copy only the needed elements, `zalloc`, `zfree` and `opaque`. But I guess it's mostly a matter of coding preference.